addons/analytic/models/analytic_plan.py

```diff
@@ -0,0 +1,126 @@
+    company_id = fields.Many2one('res.company', string='Company', default=lambda self: self.env.company)
+    account_ids = fields.One2many('account.analytic.account', 'plan_id', string="Accounts")
+    account_count = fields.Integer('Analytic Accounts Count', compute='_compute_analytic_account_count')
+    root_parent_id = fields.Many2one('account.analytic.plan', string="Root Parent", compute="_compute_root_parent",
+                                     recursive=True)
+    all_children_ids = fields.One2many('account.analytic.plan', 'root_parent_id', string="All Childrens")
+    color = fields.Integer('Color', default=_default_color)
+
+    default_applicability = fields.Selection([
+        ('optional', 'Optional'),
+        ('mandatory', 'Mandatory'),
+        ('unavailable', 'Unavailable'),
+    ], string="Default Applicability", required=True, default='optional', readonly=False)
+    applicability_ids = fields.One2many('account.analytic.applicability', 'analytic_plan_id', string='Applicability')
+
+    @api.depends('parent_id', 'parent_id.root_parent_id')
+    def _compute_root_parent(self):
+        for plan in self:
+            if plan.parent_id:
+                if plan.parent_id.root_parent_id:
+                    plan.root_parent_id = plan.parent_id.root_parent_id
+                else:
+                    plan.root_parent_id = plan.parent_id
+            else:
+                plan.root_parent_id = None
+
+    @api.depends('name', 'parent_id.complete_name')
+    def _compute_complete_name(self):
+        for plan in self:
+            if plan.parent_id:
+                plan.complete_name = '%s / %s' % (plan.parent_id.complete_name, plan.name)
+            else:
+                plan.complete_name = plan.name
+
+    @api.depends('account_ids')
+    def _compute_analytic_account_count(self):
+        for plan in self:
+            plan.account_count = len(plan.account_ids)
+
+    @api.depends('children_ids')
+    def _compute_children_count(self):
+        for plan in self:
+            plan.children_count = len(plan.children_ids)
+
+    def action_view_analytical_accounts(self):
+        result = {
+            "type": "ir.actions.act_window",
+            "res_model": "account.analytic.account",
+            "domain": [('id', 'in', self.account_ids.ids)],
+            "context": {'default_plan_id': self.id},
+            "name": _("Analytical Accounts"),
+            'view_mode': 'list,form',
+        }
+        return result
+
+    def action_view_children_plans(self):
+        result = {
+            "type": "ir.actions.act_window",
+            "res_model": "account.analytic.plan",
+            "domain": [('id', 'in', self.children_ids.ids)],
+            "context": {'default_parent_id': self.id},
+            "name": _("Analytical Plans"),
+            'view_mode': 'list,form',
+        }
+        return result
+
+    @api.model
+    def get_relevant_plans(self, **kwargs):
```

Comment:
add docstring

why the kwargs? looks like anything could be passed as argument and it makes the whole very abstract